src/git/remotes_cache.go

```diff
@@ -0,0 +1,30 @@
+package git
+
+import (
+	"errors"
+)
+
+// RemotesCache caches information about the current Git remotes.
+type RemotesCache struct {
+	remotes     []string
+	initialized bool
+}
+
+// Get provides the currently cached remotes.
+func (rc *RemotesCache) Get() ([]string, error) {
+	if !rc.initialized {
+		return rc.remotes, errors.New("cache not initialized")
```

Comment:
This feels weird to me. Why not do the initialization in this block? Would need to pass in a gitRunner or an initializer func

Reply:
This edge case never happens in real life. Just a little insurance in case the code using this cache has bugs and doesn't initialize it before reading. The actual usage of this cache is in https://github.com/git-town/git-town/blob/master/src/git/runner.go#L889-L901.